src/ProductConstructionService/ProductConstructionService.DependencyFlow/SubscriptionTriggerer.cs

```diff
@@ -145,18 +149,25 @@ await AddDependencyFlowEventAsync(
             })
             .ToList();
 
-        _logger.LogInformation("Running asset update for {subscriptionId}", _subscriptionId);
-
-        await pullRequestActor.UpdateAssetsAsync(
-            _subscriptionId,
-            subscription.SourceEnabled
-                ? SubscriptionType.DependenciesAndSources
-                : SubscriptionType.Dependencies,
-            build.Id,
-            build.GitHubRepository ?? build.AzureDevOpsRepository,
-            build.Commit,
-            assets);
-
-        _logger.LogInformation("Asset update complete for {subscriptionId}", _subscriptionId);
+        await _redisMutex.ExecuteWhenReady(
+            pullRequestActor.Id.ToString(),
+            async () =>
+            {
+                _logger.LogInformation("Running asset update for {subscriptionId}", _subscriptionId);
+
+                await pullRequestActor.UpdateAssetsAsync(
+                    _subscriptionId,
+                    subscription.SourceEnabled
+                        ? SubscriptionType.DependenciesAndSources
+                        : SubscriptionType.Dependencies,
+                    build.Id,
+                    build.GitHubRepository ?? build.AzureDevOpsRepository,
```

Comment:
I believe `build` has just `Repository` which does this. But not sure if it's the Build that's used here

Reply:
just checked, the `Client.Models.Build` has that, not this one :/

Reply:
I'll just add it and replace some places I see

Reply:
But take case - this class is auto-generated so it might not be a good idea